client/src/components/Cell/CellMain.tsx

```diff
@@ -50,6 +50,19 @@ const CellMain = ({
     [brute, randomSkill],
   );
 
+  const getRankUpIcon = useMemo(
+    () => `/images/${brute && brute.ranking === 0 ? 'ascend' : 'ranking'}.png`,
+    [brute]
+  );
+
+  const getRankUpText = useMemo(
+    () => (!brute ? '' : t(
+      brute.ranking === 0 ? 'tournamentVictoriesUntilAscend' : 'tournamentVictoriesUntilRankUp',
+      { value: getWinsNeededToRankUp(brute) }
+    )),
+    [brute, t]
+  );
```

Comment:
No need to memo these as they are not process intensive, they will clog up memory for nothing

Reply:
should i just useEffect ? not familiar with react

Reply:
No need to use anything, just put the conditional text an icon directly in the react structure